Sources/Classes/Views/BigBossButton/BigBossButtonView.swift

```diff
@@ -0,0 +1,379 @@
+
+    }
+
+    open func configureButton(withState state: BigBossButtonState) {
+        if state.contains(.enabled) {
+            button.isEnabled = true
+        }
+
+        if state.contains(.disabled) {
+            button.isEnabled = false
+        }
+
+        if state.contains(.highlighted) {
+            button.isHighlighted = true
+        }
+
+        if state.contains(.unhighlighted) {
+            button.isHighlighted = false
+        }
+
+        if state.contains(.loading) {
+            set(active: true)
+        } else {
+            set(active: false)
+        }
+    }
+}
+
+public extension BigBossButtonView {
+    struct Appearance {
+
+        var buttonFont: UIFont
+
+        var buttonStateTitles: [UIControl.State: String]
+        var buttonStateAttributtedTitles: [UIControl.State: NSAttributedString]
+        var buttonTitleStateColors: [UIControl.State: UIColor]
+        var buttonBackgroundStateColors: [UIControl.State: UIColor]
+        var buttonStateIcons: [UIControl.State: UIImage]
+
+        var buttonIconOffset: UIOffset
+        var buttonInsets: UIEdgeInsets
+
+        var buttonHeight: CGFloat
+        var buttonCornerRadius: CGFloat?
+
+        var buttonShadowPadding: CGFloat
+        var spinnerPosition: SpinnerPosition
+
+        init(buttonFont: UIFont = .systemFont(ofSize: 15),
+             buttonStateTitles: [UIControl.State: String] = [:],
+             buttonStateAttributtedTitles: [UIControl.State: NSAttributedString] = [:],
+             buttonTitleStateColors: [UIControl.State: UIColor] = [:],
+             buttonBackgroundStateColors: [UIControl.State: UIColor] = [:],
+             buttonStateIcons: [UIControl.State: UIImage] = [:],
+             buttonIconOffset: UIOffset = .zero,
+             buttonInsets: UIEdgeInsets = .zero,
+             buttonHeight: CGFloat = 50,
+             buttonCornerRadius: CGFloat? = nil,
+             buttonShadowPadding: CGFloat = 0,
+             spinnerPosition: SpinnerPosition = .center
+            ) {
+
+            self.buttonFont = buttonFont
+
+            self.buttonStateTitles = buttonStateTitles
+            self.buttonStateAttributtedTitles = buttonStateAttributtedTitles
+            self.buttonTitleStateColors = buttonTitleStateColors
+            self.buttonBackgroundStateColors = buttonBackgroundStateColors
+            self.buttonStateIcons = buttonStateIcons
+
+            self.buttonIconOffset = buttonIconOffset
+            self.buttonInsets = buttonInsets
+
+            self.buttonHeight = buttonHeight
+            self.buttonCornerRadius = buttonCornerRadius
+
+            self.buttonShadowPadding = buttonShadowPadding
+            self.spinnerPosition = spinnerPosition
+        }
+
+    }
+
+    enum SpinnerPosition {
+        case center
+        case leftToText(offset: CGFloat)
+        case rightToText(offset: CGFloat)
+    }
+}
+
+extension UIControl.State: Hashable {
+    //swiftlint:disable:next - inout_keyword
```

Comment:
don't forget to remove after https://github.com/TouchInstinct/LeadKit/pull/185 merge

Reply:
fixed